src/TieBreakingMarginComparator.php

```diff
@@ -1,18 +1,17 @@
 <?php
 namespace PivotLibre\Tideman;
 
-use Psr\Log\LoggerAwareInterface;
-use Psr\Log\LoggerAwareTrait;
+use \bitExpert\Slf4PsrLog\LoggerFactory;
 
 class TieBreakingMarginComparator
 {
 
-    use LoggerAwareTrait;
-
     private $tieBreaker;
+    private $logger;
 
     public function __construct(MarginTieBreaker $tieBreaker)
     {
+        $this->logger = LoggerFactory::getLogger(__CLASS__);
```

Comment:
I wasn't sure what to put in for the arg to this logger factory call. In Java loggers are usually created using class name as a parameter. Is that appropriate to do here too? An attractive alternative could be __PACKAGE__

Reply:
I'm not accustomed to seeing a different logger for each class. Typically you have a single logger instance for the application and it writes to a file and optionally out to stdout.

Reply:
Cool. There's a logger per class in Java, so I went with that in the first pass. Do you think the \_PACKAGE_ magic constant is the way to go then?

Reply:
Crap... i mean the \__NAMESPACE__ magic constant.